mismi-s3/test/Test/Mismi/S3.hs

```diff
@@ -66,3 +75,24 @@ withToken t f = do
   u <- liftIO $ T.pack . U.toString <$> U.nextRandom
   let a = Address b (Key . T.intercalate "/" $ ["mismi", u, unToken t])
   awsBracket_ (pure ()) (listRecursively a >>= mapM_ delete >> delete a) (f a)
+
+
+newAddress :: Token -> AWS Address
```

Comment:
Its hard because it will clean up code regardless. Should possibly just move the token on the inside do you think?